### DISCUSSION TITLE
Pass environment variables to tox _build_ environment

### DISCUSSION THREAD

**Discussion by nschloe** [tox-dev/tox#3078] [Q&A] [answered]
I'm trying to tox-test a package that needs a particular env variable _when building the package_. This config

```
[tox]
envlist = py3

[testenv]
passenv = *
; [...]
```

only seems to pass the env variables when testing. (Makes sense, it's the `[testenv]` after all.)

How can I pass environment variables to the tox _build_ process?

**Comment by jugmac00** [comment, marked as answer]
Does this help (untested)?

```
[testenv:.pkg]
pass_env = *
```

If so, I think we'd need to update our documentation.

**Reply by nschloe** [reply]
Yup, that did the trick. Thanks!

**Reply by jugmac00** [reply]
I think this is somehow sufficiently documented in https://tox.wiki/en/latest/upgrading.html#packaging-environments, so I do not see an immediate need to update our documentation. In the long term I would like to see things from the FAQ and the upgrading documentation moved or duplicated to the configuration section where appropriate.
